Intermittent failure of `test_pickle` dispatching test

I just began noticing an intermittent failure of the `test_backends.test_pickle` test. Here's the testing log:

```
$ pytest networkx/utils/tests/test_backends.py 
===================================================================== test session starts =====================================================================
platform linux -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0
Using --randomly-seed=1969482262
Matplotlib: 3.10.0
Freetype: 2.6.1
rootdir: /home/rossbar/repos/networkx
configfile: pyproject.toml
plugins: cov-6.0.0, xdist-3.6.1, randomly-3.16.0, mpl-0.17.0
collected 9 items                                                                                                                                             

networkx/utils/tests/test_backends.py ....Fss.s                                                                                                         [100%]

========================================================================== FAILURES ===========================================================================
_________________________________________________________________________ test_pickle _________________________________________________________________________

    def test_pickle():
        count = 0
        for name, func in nx.utils.backends._registered_algorithms.items():
            pickled = pickle.dumps(func.__wrapped__)
            assert pickle.loads(pickled) is func.__wrapped__
            try:
                # Some functions can't be pickled, but it's not b/c of _dispatchable
>               pickled = pickle.dumps(func)
E               AttributeError: Can't get local object 'test_not_implemented_by_nx.<locals>._stub_func'

networkx/utils/tests/test_backends.py:26: AttributeError
=================================================================== short test summary info ===================================================================
FAILED networkx/utils/tests/test_backends.py::test_pickle - AttributeError: Can't get local object 'test_not_implemented_by_nx.<locals>._stub_func'
=========================================================== 1 failed, 5 passed, 3 skipped in 0.12s ============================================================
```

The fact that it's intermittent leads me to believe this has to do with test setup via randomized test order (via pytest-randomly) rather than a real bug.

Some basic environment info:

Python version: 3.13.1
networkx hash: f2c43bc6
pytest version: 8.3.4

--- a/networkx/utils/tests/test_backends.py
+++ b/networkx/utils/tests/test_backends.py
@@ -8,6 +8,11 @@ sp = pytest.importorskip("scipy")
 pytest.importorskip("numpy")
 
 
+@nx._dispatchable(implemented_by_nx=False)
+def _stub_func(G):
+    raise NotImplementedError("_stub_func is a stub")
+
+
 def test_dispatch_kwds_vs_args():
     G = nx.path_graph(4)
     nx.pagerank(G)
@@ -171,10 +176,6 @@ def test_fallback_to_nx():
 
 
 def test_not_implemented_by_nx():
-    @nx._dispatchable(implemented_by_nx=False)
-    def _stub_func(G):
-        raise NotImplementedError("_stub_func is a stub")
-
     assert "networkx" in nx.pagerank.backends
     assert "networkx" not in _stub_func.backends
 

